packages/mui-material/src/Select/SelectInput.js

```diff
@@ -155,6 +156,7 @@ const SelectInput = React.forwardRef(function SelectInput(props, ref) {
 
     if (node) {
       setDisplayNode(node);
+      setAnchorElement(node.parentNode);
```

Comment:
`anchorElement` isn't really a separate state, but can be derived from `displayNode`, so it doesn't make sense to call `useState` for it.